test(dialog): improve test coverage of dialog

Saraph1nes opened the pull request:
<!--
首先，感谢你的贡献！😄
请阅读并遵循 [TDesign 贡献指南](https://github.com/Tencent/tdesign/blob/main/docs/contributing.md)，填写以下 pull request 的信息。
PR 在维护者审核通过后会合并，谢谢！
-->

### 🤔 这个 PR 的性质是？

- [ ] 日常 bug 修复
- [ ] 新特性提交
- [ ] 文档改进
- [ ] 演示代码改进
- [ ] 组件样式/交互改进
- [ ] CI/CD 改进
- [ ] 重构
- [ ] 代码风格优化
- [x] 测试用例
- [ ] 分支合并
- [ ] 其他

### 🔗 相关 Issue

#5631 

### 💡 需求背景和解决方案

<!--
1. 要解决的具体问题。
2. 列出最终的 API 实现和用法。
3. 涉及UI/交互变动需要有截图或 GIF。
-->

### 📝 更新日志

**tdesign-vue-next**

- test(Dialog): Improve dialog test coverage

RSS1102 commented on packages/components/dialog/__tests__/dialog.hooks.test.tsx:
```diff
@@ -0,0 +1,442 @@
+    vi.clearAllMocks();
+  });
+
+  const ActionTestComponent = defineComponent({
```
There is more than one component in this file.eslint[vue/one-component-per-file](https://eslint.vuejs.org/rules/one-component-per-file.html)

Saraph1nes commented:
<img width="1336" height="154" alt="image" src="https://github.com/user-attachments/assets/43d80da6-80ad-4ae9-8796-c90979e1a3b4" />

RSS1102 committed:
test(dialog): add unit tests for dialog component

- Add dialog.test.tsx with props and events tests
- Add dialog-card.test.tsx for DialogCard component
- Add dialog.hooks.test.tsx for useAction and useSameTarget hooks
- Add dialog.plugin.test.tsx for DialogPlugin API
- Add dialog.utils.test.tsx for utility functions
- Fix null check in keyboardEnterEvent using optional chaining

RSS1102 committed:
test(dialog): 优化 DialogCard 测试用例

RSS1102 committed:
test(dialog): 优化 Dialog 插件测试用例
